packages/dev/serializers/src/USDZ/USDZSerializer.ts

```diff
@@ -0,0 +1,732 @@
+            const xRotationOffset = Math.sin(rotation);
+            const yRotationOffset = Math.cos(rotation);
+
+            // texture coordinates start in the opposite corner, need to correct
+            offset.y = 1 - offset.y - repeat.y;
+
+            // turns out QuickLook is buggy and interprets texture repeat inverted/applies operations in a different order.
+            // Apple Feedback: 	FB10036297 and FB11442287
+            if (quickLookCompatible) {
+                // This is NOT correct yet in QuickLook, but comes close for a range of models.
+                // It becomes more incorrect the bigger the offset is
+                offset.x = offset.x / repeat.x;
+                offset.y = offset.y / repeat.y;
+
+                offset.x += xRotationOffset / repeat.x;
+                offset.y += yRotationOffset - 1;
+            } else {
+                // results match glTF results exactly. verified correct in usdview.
+                offset.x += xRotationOffset * repeat.x;
+                offset.y += (1 - yRotationOffset) * repeat.y;
+            }
+
+            return `
+            def Shader "PrimvarReader_${mapType}"
+            {
+                uniform token info:id = "UsdPrimvarReader_float2"
+                float2 inputs:fallback = (0.0, 0.0)
+                token inputs:varname = "${uv}"
+                float2 outputs:result
+            }
+            def Shader "Transform2d_${mapType}"
+            {
+                uniform token info:id = "UsdTransform2d"
+                token inputs:in.connect = </Materials/Material_${material.uniqueId}/PrimvarReader_${mapType}.outputs:result>
+                float inputs:rotation = ${(rotation * (180 / Math.PI)).toFixed(USDZExport.Precision)}
+                float2 inputs:scale = ${USDZExport._BuildVector2(repeat)}
+                float2 inputs:translation = ${USDZExport._BuildVector2(offset)}
+                float2 outputs:result
+            }
+            def Shader "Texture_${texture.uniqueId}_${mapType}"
+            {
+                uniform token info:id = "UsdUVTexture"
+                asset inputs:file = @textures/Texture_${id}.png@
+                float2 inputs:st.connect = </Materials/Material_${material.uniqueId}/Transform2d_${mapType}.outputs:result>
+                ${color !== undefined ? "float4 inputs:scale = " + USDZExport._BuildColor4(color) : ""}
+                token inputs:sourceColorSpace = "${"sRGB" /* TODO CHECK THIS */}"
+                token inputs:wrapS = "${WRAPPINGS[texture.wrapU] /* TODO CHECK THIS */}"
+                token inputs:wrapT = "${WRAPPINGS[texture.wrapV] /* TODO CHECK THIS */}"
+                float outputs:r
+                float outputs:g
+                float outputs:b
+                float3 outputs:rgb
+                ${material.needAlphaTesting() ? "float outputs:a" : "" /* TODO NEED A BETTER CHECK */}

```

Comment:
`material.needAlphaTesting() || material.needAlphaBlending()` instead of `material.needAlphaTesting()`?

Reply:
That might work.